lib/api/apiUtils/object/versioning.js

```diff
@@ -155,17 +186,17 @@ export function versioningPreprocessing(bucketName, bucketMD, objectKey,
             // overwrite the existing version, make new version null
             options.versionId = '';
             options.isNull = true;
-            options.deleteData = true;
+            options.dataToDelete = mstObjLocation;
             return callback(null, options);
         }
         // TODO check AWS behaviour
         options.versionId = reqVersionId;
-        options.deleteData = true;
+        options.dataToDelete = mstObjLocation;
```

Comment:
I think in this case (as well as on line 199) we may need to retrieve the data location of the specified version and delete it, not the mstObjLocation. Although at the current point, there is no way to PUT to a specific version ID. @ttvi can you confirm that sounds right?